codegen/cmd_test.go

```diff
@@ -40,6 +41,89 @@ var _ = Describe("Cmd", func() {
 		"encoding/protobuf/cue/cue.proto",
 	}
 
+	It("can run operator as sidecar", func() {
+		cmd := &Command{
+			// ManifestRoot: "codegen/test/chart",
+			Chart: &Chart{
+				Operators: []Operator{
+					{
+						Name: "glooMgmtServer",
+						Rbac: []rbacv1.PolicyRule{
+							{
+								Verbs:     []string{"*"},
+								APIGroups: []string{"coordination.k8s.io"},
+								Resources: []string{
+									"leases",
+								},
+							},
+						},
+						Deployment: Deployment{
+							Container: Container{
+								Image: Image{
+									Registry:   "gcr.io/gloo-mesh",
+									Repository: "gloo-mesh-mgmt-server",
+									Tag:        "0.0.1",
+								},
+								VolumeMounts: []v1.VolumeMount{{
+									Name:      "license-keys",
+									MountPath: "/etc/gloo-mesh/license-keys",
+									ReadOnly:  true,
+								}},
+							},
+							Volumes: []v1.Volume{{
+								Name: "license-keys",
+								VolumeSource: v1.VolumeSource{
+									Secret: &v1.SecretVolumeSource{
+										SecretName: "license-keys",
+									},
+								},
+							}},
+						},
+					},
+					{
+						Name: "glooAgent",
+						Rbac: []rbacv1.PolicyRule{
+							{
+								Verbs:     []string{"*"},
+								APIGroups: []string{"apiextensions.k8s.io"},
+								Resources: []string{
+									"customresourcedefinitions",
+								},
+							},
+						},
+						Deployment: Deployment{
+							Container: Container{
+								Args: []string{
+									"run", "agent",
+								},
+								Image: Image{
+									Registry:   "gcr.io/gloo-mesh",
+									Repository: "gloo-mesh-agent",
+									Tag:        "0.0.1",
+								},
+								VolumeMounts: []v1.VolumeMount{{
+									Name: "other-volume",
+								}},
+							},
+							// Should merge these with parent operator
+							Volumes: []v1.Volume{{
+								Name: "other-volume",
+							}},
+						},
+						// Indicate this operator should be deployed as a sidecar
+						// of the glooMgmtServer operator
+						ParentOperator: "glooMgmtServer",
+					},
+				},
+			},
+		}
+
+		if err := cmd.Execute(); err != nil {
```

Comment:
I tested these changes via `go get` the commit directly into gloo-mesh-enterprise, I figure we can also have a unit test here but not sure where it should be placed

@josh-pritchard